src/Google/Cache/File.php

```diff
@@ -155,7 +155,7 @@ private function getCacheDir($file, $forWrite)
     // trim the directory separator from the path to prevent double separators
     $storageDir = rtrim($this->path, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $dirHash;
     if ($forWrite && ! is_dir($storageDir)) {
-      if (! mkdir($storageDir, 0700, true)) {
+      if (!mkdir($storageDir, 0777, true)) {
```

Comment:
0777 seems little bit dangerous. What kind of data will be in the cache? Is any sensitive data there? Does it make sense to use 0770 instead?